kornia/contrib/image_prompter.py

```diff
@@ -0,0 +1,374 @@
+from __future__ import annotations
+
+from typing import Any
+
+import torch
+
+from kornia.augmentation import LongestMaxSize
+from kornia.augmentation.container.augment import AugmentationSequential
+from kornia.contrib.models import Prompts, SegmentationResults
+from kornia.contrib.models.sam import Sam, SamConfig
+from kornia.core import Tensor, pad, tensor
+from kornia.core.check import KORNIA_CHECK, KORNIA_CHECK_IS_TENSOR, KORNIA_CHECK_SHAPE
+from kornia.enhance import normalize
+from kornia.geometry.boxes import Boxes
+from kornia.geometry.keypoints import Keypoints
+
+
+class ImagePrompter:
+    """This class allow the user to run multiple query with multiple prompts for a model.
+
+    At the moment, we just support the SAM model. The model is loaded based on the given config.
+
+    For default the images are transformed to have their long side with size of the `image_encoder.img_size`. This
+    Prompter class ensure to transform the images and the prompts before prediction. Also, the image is passed
+    automatically for the method `preprocess_image`, which is responsible for normalize the image and pad it to have
+    the right size for the SAM model :math:`(\text{image_encoder.img_size}, \text{image_encoder.img_size})`. For
+    default the image is normalized by the mean and standard deviation of the SAM dataset values.
+
+    Args:
+        config: A model config to generate the model. Now just the SAM model is supported.
+        device: The desired device to use the model.
+        dtype: The desired dtype to use the model.
+
+    Example:
+        >>> # prompter = ImagePrompter() # Will load the vit h for default
+        >>> # You can load a custom SAM type for modifying the config
+        >>> prompter = ImagePrompter(SamConfig('vit_b'))
+        >>> image = torch.rand(3, 25, 30)
+        >>> prompter.set_image(image)
+        >>> boxes = Boxes(
+        ...    torch.tensor(
+        ...         [[[[0, 0], [0, 10], [10, 0], [10, 10]]]],
+        ...         device=prompter.device,
+        ...         dtype=torch.float32
+        ...    ),
+        ...    mode='xyxy'
+        ... )
+        >>> prediction = prompter.predict(boxes=boxes)
+        >>> prediction.logits.shape
+        torch.Size([1, 3, 256, 256])
+    """
+
+    def __init__(
+        self,
+        config: SamConfig = SamConfig(model_type='vit_h'),
```

Comment:
maybe we should provide it hardcoded for the Prompter can be used without Config
```suggestion
        config: SamConfig = SamConfig(model_type='vit_h', checkpoint='https://dl.fbaipublicfiles.com/segment_anything/sam_vit_h_4b8939.pth'),
```

Reply:
anything that by default you get a working model. If the user wants to use a specific model should be able to provide.